agent/engine/docker_task_engine.go

```diff
@@ -1278,6 +1295,15 @@ func (engine *DockerTaskEngine) buildCNIConfigFromTaskContainer(
 
 	cniConfig.ContainerPID = strconv.Itoa(containerInspectOutput.State.Pid)
 	cniConfig.ContainerID = containerInspectOutput.ID
+	cniConfig.ContainerNetNS = ""
+
+	// For pause containers, NetNS would be none
+	// For other containers, NetNS would be of format container:<pause_container_ID>
+	if containerInspectOutput.HostConfig.NetworkMode.IsNone() {
+		cniConfig.ContainerNetNS = containerInspectOutput.HostConfig.NetworkMode.NetworkName()
+	} else if containerInspectOutput.HostConfig.NetworkMode.IsContainer() {
+		cniConfig.ContainerNetNS = fmt.Sprintf("container:%s", containerInspectOutput.HostConfig.NetworkMode.ConnectedContainer())
+	}
```

Comment:
we should return an error here if previous two if checks both return false, since that's unexpected